From 2d70fe6248591bda75e7fba2d07cd80e3eec3d72 Mon Sep 17 00:00:00 2001
From: Xaver Hugl <xaver.hugl@kde.org>
Date: Thu, 22 May 2025 01:03:38 +0200
Subject: [PATCH] plugins/colorpicker: use BPC when converting to sRGB

It's how most applications present to the screen; without BPC the roundtrip of
picking an sRGB color off the screen may not work out

CCBUG: 491633
---
 src/plugins/colorpicker/colorpicker.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/colorpicker/colorpicker.cpp b/src/plugins/colorpicker/colorpicker.cpp
index dd9204f0af1..77ebd1d29bb 100644
--- a/src/plugins/colorpicker/colorpicker.cpp
+++ b/src/plugins/colorpicker/colorpicker.cpp
@@ -69,7 +69,7 @@ void ColorPickerEffect::paintScreen(const RenderTarget &renderTarget, const Rend
         OpenGlContext *context = effects->openglContext();
 
         context->glReadnPixels(texturePosition.x(), renderTarget.size().height() - texturePosition.y() - PIXEL_SIZE, PIXEL_SIZE, PIXEL_SIZE, GL_RGBA, GL_FLOAT, sizeof(float) * data.size(), data.data());
-        QVector3D sRGB = 255 * renderTarget.colorDescription().mapTo(QVector3D(data[0], data[1], data[2]), ColorDescription::sRGB, RenderingIntent::RelativeColorimetric);
+        QVector3D sRGB = 255 * renderTarget.colorDescription().mapTo(QVector3D(data[0], data[1], data[2]), ColorDescription::sRGB, RenderingIntent::RelativeColorimetricWithBPC);
         QDBusConnection::sessionBus().send(m_replyMessage.createReply(QColor(sRGB.x(), sRGB.y(), sRGB.z())));
         setPicking(false);
         m_scheduledPosition = QPoint(-1, -1);
-- 
GitLab

